suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +60,74 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
+        baseproduct = get_baseproduct(products_metadata)
+        if baseproduct.count != 1:
+            if not baseproduct.count:
+                message = 'There is no baseproduct'
+                log.error(message)
+                raise DistMigrationProductSetupException(message)
+            else:
+                message = ('Found multiple product definitions'
+                           'without element <flavor>')
+                log.error(message)
+                raise DistMigrationProductSetupException(message)
+        log.info('Updating Base Product to be suitable for migration')
+        update_baseproduct(baseproduct.prod_filename)
+    except Exception as issue:
+        message = 'Base Product update failed with {0}'.format(issue)
+        log.error(message)
+        raise DistMigrationProductSetupException(message)
+
+
+def get_baseproduct(products_metadata):
+    base_product_type = namedtuple(
+        'base_product_type', ['count', 'prod_filename']
+    )
+    prod_filenames = glob.glob(
+        os.path.join(products_metadata, '*.prod')
+    )
+    count = 0
+    name = None
+    try:
+        xml = ElementTree()
+        for prod_filename in prod_filenames:
+            xml.parse(prod_filename)
             register_sections = xml.findall('register')
             for register in register_sections:
-                target_sections = register.findall('target')
-                for target in target_sections:
-                    register.remove(target)
-            xml.write(
-                baseproduct, encoding="UTF-8", xml_declaration=True
-            )
+                flavor = register.findall('flavor')
+                if not flavor:
+                    count = count + 1
+                    name = prod_filename
+
+        return base_product_type(count, name)
     except Exception as issue:
-        message = 'Base Product update failed with {0}'.format(issue)
+        message = ('Getting base product for product '
+                   'update failed with {0}'.format(issue))
+        log.error(message)
+        raise DistMigrationProductSetupException(message)
+
+
+def update_baseproduct(baseproduct_prod_filename):
```

Comment:
This is a bit nitpick but when I wrote that code it was not part of a function and there was a long comment block at the beginning describing what we do here. You moved the code into a method which is good because more code has been added and that requires a bit more structure. But you named the method 'update_baseproduct' and I think that's misleading now, because what we actually do is we delete the registered target from the baseproduct. I think we should change the name to be more explicit e.g

```python
def delete_target_registration(products_metadata):
```

Thoughts ?

Reply:
Agree, it reads better with that name